karate-core/src/main/java/com/intuit/karate/StepDefs.java

```diff
@@ -43,309 +33,327 @@
-            String message = "status code was: " + response.getStatus() + ", expected: " + status
-                    + ", response time: " + responseTime + ", url: " + response.getUri() + ", response: " + rawResponse;
-            logger.error(message);
-            throw new KarateException(message);
-        }
-    }
-
-    private static MatchType toMatchType(String each, String only, boolean contains) {
-        if (each == null) {
-            if (contains) {
-                return only == null ? MatchType.CONTAINS : MatchType.CONTAINS_ONLY;
-            } else {
-                return MatchType.EQUALS;
-            }
-        } else {
-            if (contains) {
-                return MatchType.EACH_CONTAINS;
-            } else {
-                return MatchType.EACH_EQUALS;
-            }
-        }
-    }
-
-    @Then("^match (each )?([^\\s]+)( [^\\s]+)? ==$")
-    public void matchEqualsDocString(String each, String name, String path, String expected) {
-        matchEquals(each, name, path, expected);
-    }
-
-    @Then("^match (each )?([^\\s]+)( [^\\s]+)? contains( only)?$")
-    public void matchContainsDocString(String each, String name, String path, String only, String expected) {
-        matchContains(each, name, path, only, expected);
-    }
-
-    @Then("^match (each )?([^\\s]+)( [^\\s]+)? == (.+)")
-    public void matchEquals(String each, String name, String path, String expected) {
-        MatchType mt = toMatchType(each, null, false);
-        matchNamed(mt, name, path, expected);
-    }
-
-    @Then("^match (each )?([^\\s]+)( [^\\s]+)? contains( only)?(.+)")
-    public void matchContains(String each, String name, String path, String only, String expected) {
-        MatchType mt = toMatchType(each, only, true);
-        matchNamed(mt, name, path, expected);
-    }
-
-    public void matchNamed(MatchType matchType, String name, String path, String expected) {
-        AssertionResult ar = Script.matchNamed(matchType, name, path, expected, context);
-        handleFailure(ar);
-    }
-
-    @Then("^set ([^\\s]+)( .+)? =$")
-    public void setByPathDocString(String name, String path, String value) {
-        setNamedByPath(name, path, value);
-    }
-
-    @Then("^set ([^\\s]+)( .+)? = (.+)")
-    public void setByPath(String name, String path, String value) {
-        setNamedByPath(name, path, value);
-    }
-
-    public void setNamedByPath(String name, String path, String value) {
-        Script.setValueByPath(name, path, value, context);
-    }
-
-    @Given("^call ([^\\s]+)( .*)?")
-    public final void callAndUpdateVars(String name, String arg) {
-        Script.callAndUpdateVarsIfMapReturned(name, arg, context);
-    }
-
-    private void handleFailure(AssertionResult ar) {
-        if (!ar.pass) {
-            logger.error("{}", ar);
-            throw new KarateException(ar.message);
-        }
-    }
+	private static final Logger logger = LoggerFactory.getLogger(StepDefs.class);
```

Comment:
for some reason, git is not able to do a proper diff for this commit. it will be better to get that sorted out else we lose history.